### PR TITLE
feat: add `https` scheme for omitted endpoint urls

### LINKED ISSUE #7840: COPY: Apply scheme https as default for endpoint url
**Summary**

Allow query like:

```sql
COPY INTO default.test FROM 's3://helloworld/databend-autoloader-e2e-test' CONNECTION = (ENDPOINT_URL = 's3.amazonaws.com');
```

If users input a URL without `scheme`, use `https` as default.

### PR DESCRIPTION
**Pull request by ClSlaid**
Signed-off-by: ClSlaid <cailue@bupt.edu.cn>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Now users can omit the `https://` prefix in endpoint URLs.

Fixes #7840

### CODE REVIEW

**Comment by Xuanwo** [comment]
How about handle this during parser?

**Comment by ClSlaid** [comment]
> How about handle this during parse `UriLocation`?

It's not that convenient to add tests.

**Comment by Xuanwo** [comment]
> But not very convenient to test.

?

File: src/query/ast/src/parser/statement.rs [hunk 1]
```diff
@@ -1425,6 +1425,16 @@ pub fn uri_location(i: Input) -> IResult<UriLocation> {
             conn.extend(credentials_opt.map(|v| v.2).unwrap_or_default());
             conn.extend(encryption_opt.map(|v| v.2).unwrap_or_default());
 
+            // adding an `https` scheme to `endpoint_url`s if their schemes are omitted.
+            //
+            // WARNING: if there is an `endpoint_url` option in your statement to implement, please be awared.
+            if let Some(endpoint) = conn.get("endpoint_url").take() {
+                // cannot parse with `Uri` here, parsing schemeless URIs will lead to an error.
```
Comment (Xuanwo) [inline src/query/ast/src/parser/statement.rs:1432]: Why?

**Comment by ClSlaid** [comment]
This will be done in OpenDAL.

**Pull request closed without merging**